File: remote/src/main/java/com/kelsos/mbrc/mappers/DeviceSettingsMapper.java
package com.kelsos.mbrc.mappers;

import com.kelsos.mbrc.domain.DeviceSettings;
import com.kelsos.mbrc.dto.DiscoveryResponse;

public class DeviceSettingsMapper {
  public static DeviceSettings fromResponse(DiscoveryResponse response) {
    DeviceSettings settings = new DeviceSettings();
    settings.setAddress(response.getAddress());
    settings.setName(response.getName());
    settings.setPort(response.getPort());
    settings.setHttp(response.getHttp());
    return settings;
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/mappers/DeviceSettingsMapper.kt
package com.kelsos.mbrc.mappers

import com.kelsos.mbrc.data.dao.ConnectionSettings
import com.kelsos.mbrc.dto.DiscoveryResponse

object DeviceSettingsMapper {
  fun fromResponse(response: DiscoveryResponse): ConnectionSettings {
    val settings = ConnectionSettings()
    settings.address = response.address!!
    settings.name = response.name!!
    settings.port = response.port
    return settings
  }
}
